### PR TITLE
Sandy Bridge Identification Optimzations

### PR DESCRIPTION
**Pull request by markhaywood**

### CODE REVIEW

File: providers/mlx5/mlx5.c [hunk 1]
```diff
@@ -388,44 +377,46 @@ int mlx5_destroy_psv(struct mlx5_psv *psv)
...
+	if (family_id == 15)
+		family_id = ext_family_id + family_id;
 
-	fclose(fd);
-	return rc;
+	if (family_id == 6 &&
+		(model_id == 0x2A || model_id == 0x2D))
+	return 1;
+#endif /* defined(__i386__) */
+	return 0;
```
Comment (yishaih) [inline providers/mlx5/mlx5.c:419]: Do you say that on *non* i386 we may never have the Sandy Bridge CPU model ?

File: providers/mlx5/mlx5.c [hunk 2]
```diff
@@ -388,44 +377,46 @@ int mlx5_destroy_psv(struct mlx5_psv *psv)
...
-		break;
-	}
+	if (family_id == 15)
+		family_id = ext_family_id + family_id;
 
-	fclose(fd);
-	return rc;
+	if (family_id == 6 &&
+		(model_id == 0x2A || model_id == 0x2D))
+	return 1;
```
Comment (yishaih) [inline providers/mlx5/mlx5.c:417]: This line missing indentation

File: providers/mlx5/mlx5.c [hunk 3]
```diff
@@ -388,44 +377,46 @@ int mlx5_destroy_psv(struct mlx5_psv *psv)
...
+	if (family_id == 6 || family_id == 15)
+		model_id = (ext_model_id << 4) + model_id;
 
-		/* if this is a Sandy Bridge CPU */
-		if ((cur_cpu_family == 6) &&
-		    (cur_cpu_model == 0x2A || (cur_cpu_model == 0x2D) ))
-			rc = 1;
-		break;
-	}
+	if (family_id == 15)
```
Comment (yishaih) [inline providers/mlx5/mlx5.c:412]: Why do we need those lines if you return 1 in line 417 only when family_id == 6 ? if we have here 15 the below line 413 may not end-up with 6, isn't it ?

File: providers/mlx5/mlx5.c [hunk 4]
```diff
@@ -388,44 +377,46 @@ int mlx5_destroy_psv(struct mlx5_psv *psv)
...
-			if ((cur_cpu_family < 0) && (!read_number_from_line(line, &value)))
-				cur_cpu_family = value;
-		} else if (!strncmp(line, "model", 5)) {
-			if ((cur_cpu_model < 0) && (!read_number_from_line(line, &value)))
-				cur_cpu_model = value;
-		}
+	ext_family_id = EXTRACT_BITS(ax, 27, 20);
+	ext_model_id = EXTRACT_BITS(ax, 19, 16);
+	family_id = EXTRACT_BITS(ax, 11, 8);
+	model_id = EXTRACT_BITS(ax, 7, 4);
```
Comment (yishaih) [inline providers/mlx5/mlx5.c:407]: Which manual did you take those hard-coded values from ?

File: providers/mlx5/mlx5.c [hunk 5]
```diff
@@ -388,44 +377,46 @@ int mlx5_destroy_psv(struct mlx5_psv *psv)
...
+#if defined(__i386__)
+	unsigned int ax, bx, cx, dx;
+	unsigned int ext_family_id, ext_model_id, family_id, model_id;
 
-	fd = fopen("/proc/cpuinfo", "r");
-	if (!fd)
+	if (!__get_cpuid(1, &ax, &bx, &cx, &dx))
 		return 0;
 
-	while (fgets(line, 128, fd)) {
```
Comment (yishaih) [inline providers/mlx5/mlx5.c:405]: This code is called once upon driver startup and is here for years, what is the motivation just now to optimize it ?

**Comment by rleon** [comment]
Closed due to missing activity.

**Pull request closed without merging**